src/oc/lib/api/common.clj

```diff
@@ -187,17 +187,49 @@
    If no Authorization headers are supplied return nil."
   [headers passphrase]
   (when-let [token (get-token headers)]
-    (if (jwt/valid? token passphrase)
-      {:jwtoken token
-       :user (:claims (jwt/decode token))}
-      {:jwtoken false})))
+    (cond
+
+     (jwt/valid? token passphrase)
+     {:jwtoken token
+      :user (:claims (jwt/decode token))}
+
+     ;; identity token
+     (and (jwt/check-token token passphrase)
+          (:id-token (:claims (jwt/decode token))))
+     {:jwtoken false :id-token token}
+
+     :default
+     {:jwtoken false})))
+
+(defn allow-id-token
+  "Allow options request. Allow jwtoken. Allow id token. Allow anonymous."
+  [ctx]
+  (cond
+
+   (= (-> ctx :request :request-method) :options)
+   true
+
+   (:jwtoken ctx)
+   true
+
+   (:id-token ctx)
+   true
+
+   (nil? (:jwtoken ctx))
+   true
+
+   :default
+   false))
+
 
 (defn allow-anonymous
   "Allow unless there is a JWToken provided and it's invalid."
   [ctx]
   (if (= (-> ctx :request :request-method) :options)
     true ; allows allow options
-    (boolean (or (nil? (:jwtoken ctx)) (:jwtoken ctx)))))
+    (boolean (or (or (nil? (:jwtoken ctx))
+                     (:id-token ctx)) ;; id token is also anonymous
+                 (:jwtoken ctx)))))
```

Comment:
Hmmm why 2 `or`s here?
Before it was or with or w/o jwt, so i think now it should be:
or w/ or w/o JWT
or w/ or w/o id-token
at least logic suggests it to me, WYT?

Reply:
This is allowing anonymous usage when an id-token is given but the resource doesn't support an id token.